node/src/bead/mod.rs

```diff
@@ -87,8 +88,47 @@ pub enum BeadResponse {
     Genesis(Vec<BeadHash>),
     // Get all beads for IBD
     GetAllBeads(Vec<Bead>),
+    // Get beads after a specific set of hashes
+    GetBeadsAfter(Vec<BeadHash>),
     // Error response
-    Error(String),
+    Error(BeadSyncError),
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum BeadSyncError {
+    GenesisMismatch,
+    Other(String),
```

Comment:
Another error code should be UnknownBead -- if the peer requests a bead we haven't seen.